### PR TITLE
Create Docker class

### PR DESCRIPTION
**Pull request by thomasyu888**

### CODE REVIEW

**Comment by thomasyu888** [comment]
I don't know if you want to have the option of passing in the path of where the Dockerfile is and have synapse build and push the image to a synapse project.  This can definitely be done, because there is a python docker client.  However, that is just another optional dependency that we will have to enforce if we allow for this

File: synapseclient/entity.py [hunk 1]
```diff
@@ -464,6 +464,18 @@ def __init__(self, name=None, properties=None, annotations=None, local_state=Non
         super(Project, self).__init__(concreteType=Project._synapse_entity_type, properties=properties,
                                       annotations=annotations, local_state=local_state, **kwargs)
 
+class Docker(Entity):
+    """
+    Represents a Docker in Synapse.
```
Comment (kdaily) [inline synapseclient/entity.py]: Represents a Docker container in Synapse.

File: synapseclient/entity.py [hunk 2]
```diff
@@ -464,6 +464,18 @@ def __init__(self, name=None, properties=None, annotations=None, local_state=Non
...
+
+    You cannot create a Synapse Docker image with this function, please follow instructions here (http://docs.synapse.org/articles/docker.html) to create one.
```
Comment (kdaily) [inline synapseclient/entity.py]: Confusing about what you are creating then to have this warning...maybe this should be shown when a user creates, and an image doesn't exist (or both here and there)?

**Comment by thomasyu888** [comment]
Figure out how to pass in username and password

**Comment by kdaily** [comment]
Use environment variables for passwords, or a config file.

**Comment by thomasyu888** [comment]
You need to log into the registry to be able to create a docker container

**Comment by thomasyu888** [comment]
There is a newer pull request done by Ziming.

**Pull request closed without merging**